examples/python/celsius_connectors/at_least_once_feed

```diff
@@ -0,0 +1,282 @@
+                self.sources[stream.id] = source
+        # stream is new to us
+        else:
+            if stream.is_open:
+                # This is an error for MultiFileReaderConnector
+                raise ConnectorError("Can't open a new source from a stream. "
+                                     "Please use the add_source interface.")
+            # nothing to do with a closed stream for which there is no source
+
+    def add_source(self, source):
+        # add to self.sources
+        _id = self.get_id(source.name)
+        self.sources[_id] = source
+        self.keys.append(_id)
+        # add to joining set so we can control the starting sequence
+        self.joining.add(_id)
+
+        # send a notify
+        self.notify(_id, source.name, source.point_of_ref())
+
+    @staticmethod
+    def get_id(text):
+        """
+        Get a positive U64 from the I64 of the hash of text
+        """
+        return ctypes.c_size_t(hash(text)).value
+
+    # Make this class an iterable:
+    def __next__(self):
+        if self.error:
+            # TODO: Maybe we should raise StopIteration here and ask the user
+            # to check `self.error` before choosing how to procede?
+            raise ConnectorError("Connector connection has been closed after "
+                "receiving an error message from Wallaroo: {}"
+                .format(self.error.message))
+        if len(self.keys) > 0:
+            try:
+                # get next position
+                self._idx = (self._idx + 1) % len(self.keys)
+                # get key of that position
+                key = self.keys[self._idx]
+                # get source at key
+                source = self.sources[key]
+                # get value from source
+                value, point_of_ref = next(source)
+                # send it as a message
+                msg = cwm.Message(
+                    stream_id = key,
+                    flags = cwm.Message.Key,
+                    message_id = point_of_ref,
+                    key = source.key,
+                    message = value)
+                self.write(msg)
+                return msg
+            except StopIteration:
+                # if the source threw a StopIteration, send an EOS message
+                # for it, then close it and remove it from sources
+                self.end_of_stream(stream_id = key)
+                source.close()
+                del self.sources[key]
+                self.keys.pop(self._idx)
+                self._idx -= 1 # to avoid skipping in the round-robin sender
+                return source
+        else:
+            raise StopIteration
+
+
+client = MultiFileReaderConnector(
+        args=None,
+        required_params=['host', 'port', 'filenames'],
+        optional_params=['timeout', 'delay'])
+
+# Open a connection with a hello message
+hello = cwm.Hello("0.0.1", "1234", "celsius at least once", "instance")
+client.initiate_handshake(hello)
+client.start()
```

Comment:
I think we should implicitly start.
This is something I want to change in terms of where the `Thread` that runs the async loop lives.
Currently it's part of the class, and terminates when the async loop exits (e.g. when `self.close()` is called, or when an error is encountered). 
But we could make it an attribute of the class, which would allow a new thread to take its place during a restart sequence, for example, without requiring the rest of the class to be copied over to a new one.